ConfigSpace/hyperparameters.pyx

```diff
@@ -1395,13 +1586,176 @@ cdef class NormalIntegerHyperparameter(IntegerHyperparameter):
+        self.bfhp = BetaFloatHyperparameter(self.name,
+                                              self.alpha,
+                                              self.beta,
+                                              log=self.log,
+                                              q=self.q,
+                                              lower=self.lower,
+                                              upper=self.upper,
+                                              default_value=self.default_value)
+
+        self.default_value = self.check_default(default_value)
+        self.normalized_default_value = self._inverse_transform(self.default_value)
+        
+    def __repr__(self) -> str:
+        repr_str = io.StringIO()
+        repr_str.write("%s, Type: BetaInteger, Alpha: %s Beta: %s, Range: [%s, %s], Default: %s" % (self.name, repr(self.alpha), repr(self.beta), repr(self.lower), repr(self.upper), repr(self.default_value)))
+        
+        if self.log:
+            repr_str.write(", on log-scale")
+        if self.q is not None:
+            repr_str.write(", Q: %s" % str(self.q))
+        repr_str.seek(0)
+        return repr_str.getvalue()
+
+    def __eq__(self, other: Any) -> bool:
+        """
+        This method implements a comparison between self and another
+        object.
+
+        Additionally, it defines the __ne__() as stated in the
+        documentation from python:
+            By default, object implements __eq__() by using is, returning NotImplemented
+            in the case of a false comparison: True if x is y else NotImplemented.
+            For __ne__(), by default it delegates to __eq__() and inverts the result
+            unless it is NotImplemented.
+
+        """
+        if not isinstance(other, self.__class__):
+            return False
+
+        return (
+            self.name == other.name and
+            self.alpha == other.alpha and
+            self.beta == other.beta and
+            self.log == other.log and
+            self.q == other.q and
+            self.lower == other.lower and
+            self.upper == other.upper
+        )
+
+    def __hash__(self):
+        return hash((self.name, self.alpha, self.beta, self.lower, self.upper, self.log, self.q))
+
+    def __copy__(self):
+        return BetaIntegerHyperparameter(
+            name=self.name,
+            default_value=self.default_value,
+            alpha=self.alpha,
+            beta=self.beta,
+            log=self.log,
+            q=self.q,
+            lower=self.lower,
+            upper=self.upper,
+            meta=self.meta
+        )
+
+    def to_uniform(self) -> 'UniformIntegerHyperparameter':
+        return UniformIntegerHyperparameter(self.name,
+                                            self.lower,
+                                            self.upper,
+                                            default_value=self.default_value,
+                                            q=self.q, log=self.log)
```

Comment:
Maybe pass the meta information here, too?

Reply:
Fixed for all parameters - they were missing for Normals, too.